qa-pipelines/tasks/run-test.sh

```diff
@@ -87,8 +91,44 @@ done
 
 pod_status=$(container_status ${TEST_NAME})
 
+if [[ ${TEST_NAME} == "acceptance-tests" ]] && [[ $pod_status -gt 0 ]]; then
+  export CATS_RERUN=1
+  while [[ $CATS_RERUN -lt 5 ]] && [[ $pod_status -gt 0 ]]; do
+    export CATS_SUITES="=$(
+        # Gets comma-separated list of all failing tests.
+        # The first tr removes the formatting control characters from the test output, because it breaks grep
+        # The sed command is required because the displayed names for docker and ssh suites are different from the variable
+        # expected by CATS to run those tests (diego_docker and diego_ssh respectively)
+        kubectl logs --namespace=scf ${TEST_NAME} \
+        | tr -d '\000\001\220\221\133\033\071\060\061\155' \
```

Comment:
I tried `tr -d [:cntrl:] first` but it didn't work for some reason. These were the bytes in the way of me grepping the things I needed to grep for

Reply:
If you need to test it, you can use `fly watch` to pipe the output from the acceptance-tests task of a build.

Reply:
Ooops, `[:cntrl:]` includes `\r`, `\n`… that seems bad :D `tr -dC '[:print:]\r\n` seems okay (but of course this leaves all the colour codes in). As is, I think your command breaks things like `echo 'Acceptance Tests Complete; exit status: 1'` (because the `m` gets deleted).

If you want to remove the colour codes, maybe try `perl -p -i -e 's@\e\[.*?m@@g'` instead?

Reply:
Ah, okay, so you're trying to remove the colouring, then look for `Fail]` and print the second bracketed expression.  So, something like
```bash
kubectl logs ...
    | grep Fail\] # look for relevant lines
    | perl -pe 's@\e.*?m@@g'  # strip colours
    | awk '{ print $2 }' # print the suite name
    | tr -d '[]'  # drop the brackets
    | sort -u # sort, uniq
```